src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+            throw new DataProcessingException("Couldn't create book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String selectQuery = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement statement = connection.prepareStatement(selectQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(mapResultSetToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String selectAllQuery = "SELECT * FROM books";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(selectAllQuery)) {
+            while (resultSet.next()) {
+                books.add(mapResultSetToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't retrieve all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateQuery = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement statement = connection.prepareStatement(updateQuery)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String deleteQuery = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement statement = connection.prepareStatement(deleteQuery)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete book by id " + id, e);
+        }
+    }
+
+    private Book mapResultSetToBook(ResultSet resultSet) throws SQLException {
+        Book book = new Book();
+        book.setId(resultSet.getLong("id"));
+        book.setTitle(resultSet.getString("title"));
+        book.setPrice(resultSet.getBigDecimal("price"));
+        return book;

```

Comment:
Good job on extracting the mapping logic to a separate method. However, consider using either setters or a constructor exclusively for initializing the `Book` object for consistency.